safe_transaction_service/history/views.py

```diff
@@ -174,6 +174,7 @@ class AllTransactionsListView(ListAPIView):
         django_filters.rest_framework.DjangoFilterBackend,
         OrderingFilter,
     )
+    ordering_fields = ["created"]
```

Comment:
The original issue reported that "ordering" doesn't work on this endpoint. This seems to just allow to set the ordering to the "created" field. Without the `ordering_fields` the following should happen:

> If you don't specify an ordering_fields attribute on the view, the filter class will default to allowing the user to filter on any readable fields on the serializer specified by the serializer_class attribute.

https://www.django-rest-framework.org/api-guide/filtering/#specifying-which-fields-may-be-ordered-against

Reply:
You are correct, but this is the serializer class:
```
serializer_class = (
    serializers.AllTransactionsSchemaSerializer
 )  `# Just for docs, not used

----------------------------------------------------------------------------------------------------

class AllTransactionsSchemaSerializer(serializers.Serializer):
    """
    Just for the purpose of documenting, don't use it
    """

    tx_type_1 = SafeModuleTransactionWithTransfersResponseSerializer()
    tx_type_2 = SafeMultisigTransactionWithTransfersResponseSerializer()
    tx_type_3 = EthereumTxWithTransfersResponseSerializer()

```
Then the fields that ordering_filter can get by default is:
```
[('tx_type_1', 'Tx type 1'), ('tx_type_2', 'Tx type 2'), ('tx_type_3', 'Tx type 3')]
```
For this reason this default without any fields aren't working.

Reply:
Maybe I'm lacking some context but you added "created" as a default `ordering_fields`. The original issue is about not being able to use `ordering` at all (not which one is the default one). In the issue they also give an example of `nonce` ordering - is it possible to do sort with the nonce now?

And with the ordering that we are adding, will it impact the service in any way (performance wise)?

Reply:
I added one field that let us order by this if is included in the request, it is not a default value. 
I meant without `ordering_fields` defined and with the serializer_class that I mentioned means that ordering is not working for any value different to `tx_type_1,..` that doesn't make sense. Adding `created` to ordering_fields indicates to ordering_filter that this field is a valid field. 
I could see that the default ordering is defined in the queryset, I think this is not affecting the performance because if the ordering parameter is not included in the request then is going to keep the queryset result. 

```
# Tricky, we merge SafeTx hashes with EthereumTx hashes
        queryset = (
            multisig_safe_tx_ids.union(erc20_tx_ids)
            .union(erc721_tx_ids)
            .union(internal_tx_ids)
            .union(module_tx_ids)
            .order_by("-execution_date", "-safe_nonce", "block", "-created")
        )
 ```

Reply:
To add default ordering to orderingFilter: 
```
[Specifying a default ordering](https://www.django-rest-framework.org/api-guide/filtering/#specifying-a-default-ordering)

If an ordering attribute is set on the view, this will be used as the default ordering.
```
but I wouldn't include because the queryset has defined the order_by.

Reply:
> I added one field that let us order by this if is included in the request, it is not a default value. I meant without `ordering_fields` defined and with the serializer_class that I mentioned means that ordering is not working for any value different to `tx_type_1,..` that doesn't make sense. Adding `created` to ordering_fields indicates to ordering_filter that this field is a valid field. I could see that the default ordering is defined in the queryset, I think this is not affecting the performance because if the ordering parameter is not included in the request then is going to keep the queryset result.
> 
> ```
> # Tricky, we merge SafeTx hashes with EthereumTx hashes
>         queryset = (
>             multisig_safe_tx_ids.union(erc20_tx_ids)
>             .union(erc721_tx_ids)
>             .union(internal_tx_ids)
>             .union(module_tx_ids)
>             .order_by("-execution_date", "-safe_nonce", "block", "-created")
>         )
> ```

Ah yes I didn't mean as "default" but that you can now use it. My bad. So if you set `created` or `-created` now it changes the outcome right?

Why not add the other fields specified in the ticket?

Reply:
yes, now changes the outcome. 
We could include fields to change the default ordering: 
`["execution_date", "safe_nonce", "block", "created"]`
What do you think?